shared/src/DatadogSymbolsServer/DotnetApmSymbolsCache.cs

```diff
@@ -0,0 +1,227 @@
+
+            List<(string RelativePath, Guid Guid, uint Age)> pdbFilesInfo = [];
+
+            foreach (var image in Directory.EnumerateFiles(binariesPath, "*.dll", SearchOption.AllDirectories))
+            {
+                var parent = Path.GetFileName(Path.GetDirectoryName(image));
+                if (string.IsNullOrWhiteSpace(parent))
+                {
+                    continue;
+                }
+
+                var peHeader = new PeNet.PeFile(image);
+                if (peHeader.ImageDebugDirectory != null)
+                {
+                    foreach (var d in peHeader.ImageDebugDirectory)
+                    {
+                        if (d != null && d.CvInfoPdb70 != null)
+                        {
+                            pdbFilesInfo.Add((Path.Combine(parent, GetWindowsFilename(d.CvInfoPdb70.PdbFileName)), d.CvInfoPdb70.Signature, d.CvInfoPdb70.Age));
+                        }
+                    }
+                }
+            }
+
+            var symbolsPath = Path.Combine(tmpPath, "symbols");
+
+            using var symbolsRequest = await httpClient.GetAsync($"DataDog/dd-trace-dotnet/releases/download/v{version}/windows-native-symbols.zip", cancellationToken);
+
+            await using Stream symbolsStream = await symbolsRequest.Content.ReadAsStreamAsync(cancellationToken);
+            ExtractTo(symbolsPath, symbolsStream);
+
+            foreach (var fileInfo in pdbFilesInfo)
+            {
+                var pdbPath = Path.Combine(symbolsPath, fileInfo.RelativePath);
+                if (!File.Exists(pdbPath))
+                {
+                    continue;
+                }
+
+                var guidd = fileInfo.Guid.ToString("N").ToLower() + fileInfo.Age.ToString();
+
+                var destFolder = Path.Combine(_rootPath, guidd);
+                var newfile = Path.Combine(destFolder, Path.GetFileName(fileInfo.RelativePath));
+                Directory.CreateDirectory(destFolder);
+
+                File.Copy(pdbPath, newfile, overwrite: true);
+            }
+
+            Directory.Delete(tmpPath, recursive: true);
+
+            void ExtractTo(string destPath, Stream stream)
+            {
+                using var zip = new ZipArchive(stream);
+                Directory.CreateDirectory(destPath);
+                zip.ExtractToDirectory(destPath, overwriteFiles: true);
+            }
+
+            string GetWindowsFilename(string filePath)
+            {
+                var i = filePath.LastIndexOf('\\');
+                if (i == -1)
+                {
+                    return filePath;
+                }
+
+                return filePath.Substring(i + 1);
+            }
+        }
+
+        public async Task Initialize(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Starting symbols cache");
+
+            string[] versions = ["3.10.1", "3.8.0", "3.9.1", "3.9.0", "3.6.0", "2.61.0", "2.56.0"];
```

Comment:
Why tare these versions special? 🤔  Should we always ingest the latest version (as that's what we tend to check?)